supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/GoSlowJobSubmitActor.scala

```diff
@@ -0,0 +1,71 @@
+package cromwell.backend.impl.aws
+
+import java.util.concurrent.Executors
+
+import akka.actor.{Actor, ActorLogging, Props}
+import cats.effect.{IO, Timer}
+import cromwell.backend.impl.aws.GoSlowJobSubmitActor.{CheckForWork, SubmitForMe}
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.batch.BatchClient
+import software.amazon.awssdk.services.batch.model.SubmitJobResponse
+
+import scala.concurrent.{ExecutionContext, Promise}
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+class GoSlowJobSubmitActor(configRegion: Option[Region]) extends Actor with ActorLogging {
+
+  implicit val ec: ExecutionContext = context.dispatcher
+  implicit val timer: Timer[IO] = cats.effect.IO.timer(ExecutionContext.fromExecutor(Executors.newSingleThreadScheduledExecutor))
+
+  val WorkInterval = 100.millis
+
+  lazy val client = {
+    val builder = BatchClient.builder()
+    configRegion.foreach(builder.region)
+    builder.build
+  }
+
+  // Maps job ID to status
+  var workQueue: Vector[SubmitForMe] = Vector.empty
```

Comment:
Should this be to prevent outside influences modifying the state inside the actor?